uber/models/department.py

```diff
@@ -407,6 +416,38 @@ def hours(self):
     def end_time(self):
         return self.start_time + timedelta(hours=self.duration)
 
+    def working_limit_ok(self, attendee):
+        attendee_hour_map = attendee.hour_map
+        hours_worked = self.duration
+        working_hour_limit = self.department_max_consecutive_hours
+        if working_hour_limit == 0:
+            working_hour_limit = 1000 # just default to something large
+
+        # count the number of filled hours before this shift
+        cur = self.start_time - timedelta(hours=1)
+        while cur in attendee_hour_map:
+            hours_worked += 1
+            this_job_hour_limit = attendee_hour_map[cur].department_max_consecutive_hours
+            if this_job_hour_limit > 0:
+                working_hour_limit = min(working_hour_limit, this_job_hour_limit)
+            cur = cur - timedelta(hours=1)
+
+        # count the number of filled hours after this shift
+        cur = self.start_time + timedelta(hours=self.duration)
+        while cur in attendee_hour_map:
+            hours_worked += 1
+            this_job_hour_limit = attendee_hour_map[cur].department_max_consecutive_hours
+            if this_job_hour_limit > 0:
+                working_hour_limit = min(working_hour_limit, this_job_hour_limit)
+            cur = cur + timedelta(hours=1)
+
+        # working_hour_limit is the *min* of Department.max_consecutive_hours
```

Comment:
This is a great comment! Mind moving it to the top of the function in a triple-string block? It does a great job of explaining what this function is for overall.